packages/smart-contracts/scripts-create2/xdeployer.ts

```diff
@@ -70,14 +71,22 @@ export const xdeploy = async (
     let receipt = undefined;
     let deployed = false;
     let error = undefined;
-    const gasPrice = await provider.getGasPrice();
+
+    const txOverrides: Overrides = await utils.calculateGasFees(provider as any, true);
```

Comment:
The `suggestFee` lib needs a `JsonRpcProvider`, but our `utils` give the parent class `Provider` instead.
@olivier7delf You can use  `as providers.JsonRpcProvider`, that's what we were doing before:
https://github.com/RequestNetwork/requestNetwork/blob/121aeaaed4c8f65a57b9f20a1cf1e31d75e09d3f/packages/ethereum-storage/src/ethereum-tx-submitter.ts#L21

Reply:
OK. 
better handle the test conversion inside the util function

Reply:
maybe a `try {...} catch (...)` around `await suggestFees` could be worth it, to be sure that if it fails because of a bad provider it is handled well

Reply:
I proposed it, lmk:
`const suggestedFee = await suggestFees(
    provider instanceof providers.JsonRpcProvider
      ? provider
      : (provider as providers.JsonRpcProvider),
  );`

Reply:
@olivier7delf when transpiled to Javascript it will become:
`suggestFees(provider instanceof providers.JsonRpcProvider ? provider : provider)`
; so it's not useful to do this.

Reply:
Ok thx, this seems to solve the issue: `const suggestedFee = await suggestFees(provider as providers.JsonRpcProvider) `